### ISSUE TITLE
win_pkg: salt minion cannot be upgraded when computer is running on batteries

### ISSUE THREAD

**Issue by morganwillcock** [saltstack/salt#35943]
Since the minion service is upgraded via the Task Scheduler, the default Task options to prevent a job running if the computer is running on batteries will prevent the minion from being upgraded.

You don't get any feedback from the module:
<pre>salt salt-test pkg.install salt_minion
salt-test:
    ----------
ERROR: Minions returned with non-zero exit code</pre>
...but you can see the problem by checking the Task history:
> Task Scheduler did not launch task "\update-salt-software"  because computer is running on batteries. User Action: If launching the task on batteries is required, change the respective flag in the task configuration.

> Task Scheduler failed to start "\update-salt-software" task for user "NT AUTHORITY\System". Additional Data: Error Value: 2147750692.

This actually has the potential to be quite useful when upgrading the minion as part of an automated state, but I think there needs to be a way of indicating that the Task should not have this flag set (particularly when trying to manually upgrade using the module). Plus the module should probably try to give you some better feedback.

As a workaround you can manually run the Task.

Tested on 2016.3.2

**Comment by Ch3LL** [comment]
@morganwillcock thanks looks like we need to do a better job at notifying the user why its not updating.

**Comment by Ch3LL** [comment]
ping @twangboy fyi

**Comment by twangboy** [comment]
@Ch3LL I'll add it to my list

**Comment by morganwillcock** [comment]
@Ch3LL thank you for looking at this so quickly.

@twangboy I've submitted a PR but I've targeted the oldest stable branch (hopefully the right thing to do; after reading the contribution guide). It does make the assumption that only the minion package is installed or uninstalled by the Task Scheduler. I can't think of any reason why any other installer would need to do this, especially since you cannot check the return code for the action that was run.
